sound/soc/sof/ipc.c

```diff
@@ -446,44 +448,30 @@ void snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id)
 }
 EXPORT_SYMBOL(snd_sof_ipc_reply);
 
-static void ipc_comp_notification(struct snd_sof_dev *sdev,
-				  struct sof_ipc_cmd_hdr *hdr)
+static void ipc_comp_notification(struct snd_sof_dev *sdev, void *full_msg)
 {
+	struct sof_ipc_cmd_hdr *hdr = full_msg;
```

Comment:
I have picked `void *` precisely to avoid assumptions and a possible confusion for newcomers.

The function would assume that the `struct sof_ipc_cmd_hdr *` is pointing to the header part of a message struct they are receiving. The use of strong type does not guarantee it.
However if we say that: here is a pointer to the start of the message we have received, deal with it
It is essentially tells that the whole message is behind of the pointer, no loophole, you can not call the function and pass a pointer to a separate struct with a clean consciousness ;)  

To avoid cases like: https://github.com/thesofproject/linux/pull/3054/commits/fd79a69f6ba99e7670a07a720699d50766aa08b2#r713625534

It was assuming that the stream_tag pointer is part of the ipc struct, but it has never been guaranteed.

Reply:
would using `container_of()` after passing a header pointer make it any better?